src/config/kyuubi.py

```diff
@@ -41,9 +51,22 @@ def _auth_conf(self) -> dict[str, str]:
             "kyuubi.authentication.jdbc.query": self._get_authentication_query(),
         }
 
+    @property
+    def _ha_conf(self) -> dict[str, str]:
+        if not self.zookeeper_info:
+            return {}
+        return {
+            "kyuubi.ha.addresses": self.zookeeper_info.uris,
+            # FIXME: Get this value from self.context.zookeeper.uris when znode created by
+            # zookeeper charm has enough permissions for Kyuubi to work
+            "kyuubi.ha.namespace": HA_ZNODE_NAME,
```

Comment:
Here, as a workaround we supply a different name to DatabaseRequires and a different name to Kyuubi config so that Kyuubi can create the znode itself with the necessary permissions.

Once we figure out the ALCs that are set to znode creates by the Zookeeper charm and agree on it, we can then use the znode created by zookeeper instead.

Reply:
yes, we should probably update the relation interface to allow to create "databases" that are writable by the requirer, and readable by others. This seems a fairly common use-case we may want to have a consistent UX, and beside, since we are using the `DatabaseRequires/DatabaseProvides` classes, we should definitely discuss with the wider DP team. 

I have mentioned this to @welpaolo today that I feel it is worth bringing this up at DP Office hours